src/tools/ShapeSelect.coffee

```diff
@@ -0,0 +1,85 @@
+{Tool} = require './base'
+{createShape} = require '../core/shapes'
+
+module.exports = class ShapeSelect extends Tool
+  name: 'ShapeSelect'
+  usesSimpleAPI: false
+
+  constructor: (lc) ->
+    # This is a 'shadow' canvas -- we'll reproduce the shapes here, each shape
+    # with a different color that corresponds to their index. That way we'll
+    # be able to find which shape to select on the main canvas by pixel color.
+    @selectCanvas = document.createElement('canvas')
+    @selectCanvas.style['background-color'] = 'transparent'
+    @selectCtx = @selectCanvas.getContext('2d')
+
+  didBecomeActive: (lc) ->
```

Comment:
I thought it felt fairly straightforward. The hardest part was looking at examples and seeing a mix and remembering that this is the newer API :)